Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -167,14 +172,25 @@ namespace AzToolsFramework
 
     bool PrefabEditorEntityOwnershipService::DestroyEntity(AZ::Entity* entity)
     {
+        AZ_Assert(entity, "Tried to destroy a null entity");
```

Comment:
Will do. I thought in our coding guidelines we shouldn't be preventing crashes for programmer error/asserts, but that could have changed.

Reply:
Agree with @michabr, asserts are not needed to be protected, if the function is never meant to be called with entity null then an assert is enough and failing asserts are meant to leave the app in an unknown state because it's never meant to happen. If it's possible to happen then it's not an az_assert, but an az_error that needs to be handled.